### PR TITLE
Cleare stale route records

### PR DESCRIPTION
**Pull request by soffokl**
#2233

### CODE REVIEW

File: cmd/di.go [hunk 1]
```diff
@@ -194,6 +194,8 @@ func (di *Dependencies) Bootstrap(nodeOptions node.Options) error {
 		return err
 	}
 
+	netutil.ClearStaleRoutes()
```
Comment (anjmao) [inline cmd/di.go:197]: After we merge https://github.com/mysteriumnetwork/node/pull/2302 if would be better to move this to suppervisor command to clear routes.

File: utils/netutil/network_windows.go [hunk 2]
```diff
@@ -32,16 +32,15 @@ func assignIP(iface string, subnet net.IPNet) error {
...
-		return err
-	}
-
+func excludeRoute(ip, gw net.IP) error {
 	out, err := exec.Command("powershell", "-Command", "route add "+ip.String()+"/32 "+gw.String()).CombinedOutput()
 	return errors.Wrap(err, string(out))
 }
 
+func deleteRoute(ip, gw string) error {
+	return cmdutil.SudoExec("route", "delete", ip+"/32", "gw")
```
Comment (anjmao) [inline utils/netutil/network_windows.go:41]: This looks not correct as gw arg is not used. What is this last param "gw" ?

Comment (soffokl) [reply utils/netutil/network_windows.go]: Good catch, we don't need this this `"gw"`, removed it.

File: cmd/di.go [hunk 3]
```diff
@@ -394,6 +396,8 @@ func (di *Dependencies) bootstrapStorage(path string) error {
 
 	di.Storage = localStorage
 
+	netutil.SetRouteManagerStorage(di.Storage)
```
Comment (zolia) [inline cmd/di.go:399]: Why not to have this in factory? And call ClearStaleRoutes() on routeManager instance?

File: supervisor/daemon/daemon.go [hunk 4]
```diff
@@ -132,7 +143,15 @@ func (d *Daemon) wgDown(args ...string) (err error) {
 	if *interfaceName == "" {
 		return errors.New("-iface is required")
 	}
-	return d.monitor.Down(*interfaceName)
+
+	err = d.monitor.Down(*interfaceName)
+	if err != nil {
+		return err
```
Comment (zolia) [inline supervisor/daemon/daemon.go:149]: wrap an error

File: utils/netutil/network.go [hunk 5]
```diff
@@ -21,25 +21,96 @@ import (
...
+		}
+	}
 }
 
 // ExcludeRoute excludes given IP from VPN tunnel.
 func ExcludeRoute(ip net.IP) error {
-	return excludeRoute(ip)
+	gw, err := gateway.DiscoverGateway()
+	if err != nil {
+		return err
```
Comment (zolia) [inline utils/netutil/network.go:89]: wrap an error

File: utils/netutil/network.go [hunk 6]
```diff
@@ -21,25 +21,96 @@ import (
...
+	}
+
+	if defaultRouteManager != nil {
```
Comment (zolia) [inline utils/netutil/network.go:92]: these check would be redundant if proper struct instance is used.

File: cmd/di.go (hunk 3 above)
Comment (soffokl) [reply cmd/di.go:400]: Routes cleaning is a system-wide component. You never need to have more than one of them. Having it independent allows to access it simpler, without passing it down through all components.

**Pull request merged**